homeassistant/components/media_player/__init__.py

```diff
@@ -523,7 +521,36 @@ def source_list(self):
 
     @property
     def supported_media_commands(self):
-        """Flag media commands that are supported."""
+        """Flag media commands that are supported.
+
+        DEPRECATED: Included for temporary custom platform compatibility.
+        """
+        return None
+
+    @property
+    def supported_features(self):
+        """Flag media player features that are supported."""
+        # Begin temporary transition logic
+
+        if self.supported_media_commands is not None:
+            # If this platform is still using supported_media_commands, issue
+            # a logger warning once with instructions on how to fix it.
+            if not getattr(self, '_supported_features_warned', False):
+                import inspect
```

Comment:
I don't really think there's a better place to put it. I deferred the message generation to an executor function, so that the event loop doesn't have to wait for it. Really though, this will only get called by custom components that haven't been updated, and we tell them exactly how to fix it. This will only be temporary for a few versions anyway to give people some warning that it's going away.